cmd/zpool/zpool_main.c

```diff
@@ -2471,61 +2474,587 @@ zpool_do_import(int argc, char **argv)
+					iostat_bottom_labels[idx][i]);
+	}
 
-	for (i = 0; i < cb->cb_namewidth; i++)
-		(void) printf("-");
-	(void) printf("  -----  -----  -----  -----  -----  -----\n");
+	printf("\n");
+	__print_iostat_separator(cb, force_column_width);
 }
 
 static void
 print_iostat_header(iostat_cbdata_t *cb)
 {
-	(void) printf("%*s     capacity     operations    bandwidth\n",
-	    cb->cb_namewidth, "");
-	(void) printf("%-*s  alloc   free   read  write   read  write\n",
-	    cb->cb_namewidth, "pool");
-	print_iostat_separator(cb);
+	__print_iostat_header(cb, 0, NULL);
 }
 
+
 /*
  * Display a single statistic.
  */
 static void
-print_one_stat(uint64_t value)
+print_one_stat(uint64_t value, enum zfs_nicenum_format format,
+		unsigned int column_size, boolean_t scripted)
 {
 	char buf[64];
+	__zfs_nicenum(value, buf, sizeof (buf), format);
+
+	if (scripted)
+		printf("\t%s", buf);
+	else
+		printf("  %*s", column_size, buf);
+}
+
+/*
+ * Calculate the vdev stats
+ *
+ * Subtract oldvs from newvs, apply a scaling factor, and save the resulting
+ * stats into calcvs.
+ */
+static void
+calc_iostats(vdev_stat_t *oldvs, vdev_stat_t *newvs, vdev_stat_t *calcvs,
+		iostat_cbdata_t *cb)
+{
+	int i, j;
+	uint64_t tdelta = newvs->vs_timestamp - oldvs->vs_timestamp;
+	double scale;
+	memcpy(calcvs, newvs, sizeof (*calcvs));
+
+	if ((oldvs->vs_timestamp == 0) && (cb->cb_flags & VS_HISTO_M)) {
+		/*
+		 * If we specify printing histograms with no time interval, then
+		 * print the histogram numbers over the entire lifetime of the
+		 * vdev.
+		 */
+		scale = 1;
+	} else {
+		if (tdelta == 0)
+			scale = 1.0;
+		else
+			scale = (double)NANOSEC / tdelta;
+	}
+
+	for (j = 0; j < ARRAY_SIZE(calcvs->vs_queue_histo); j++)
+		for (i = 0; i < ARRAY_SIZE(calcvs->vs_queue_histo[0]); i++)
+			calcvs->vs_queue_histo[j][i] = scale *
+			(newvs->vs_queue_histo[j][i] -
+			oldvs->vs_queue_histo[j][i]);
+
+	for (j = 0; j < ARRAY_SIZE(calcvs->vs_disk_histo); j++)
+		for (i = 0; i < ARRAY_SIZE(calcvs->vs_disk_histo[0]); i++)
+			calcvs->vs_disk_histo[j][i] = scale *
+				(newvs->vs_disk_histo[j][i] -
+				oldvs->vs_disk_histo[j][i]);
+
+	for (j = 0; j < ARRAY_SIZE(calcvs->vs_total_histo); j++)
+		for (i = 0; i < ARRAY_SIZE(calcvs->vs_total_histo[0]); i++)
+			calcvs->vs_total_histo[j][i] = scale *
+				(newvs->vs_total_histo[j][i] -
+				oldvs->vs_total_histo[j][i]);
+
+	for (j = 0; j < ARRAY_SIZE(calcvs->vs_queue_active); j++)
+		calcvs->vs_queue_active[j] = newvs->vs_queue_active[j];
+
+	for (j = 0; j < ARRAY_SIZE(calcvs->vs_ops); j++)
+		calcvs->vs_ops[j] = scale *
+				(newvs->vs_ops[j] - oldvs->vs_ops[j]);
+
+	for (j = 0; j < ARRAY_SIZE(calcvs->vs_bytes); j++)
+		calcvs->vs_bytes[j] = scale *
+				(newvs->vs_bytes[j] - oldvs->vs_bytes[j]);
+
+}
+
+/* Find the max element in an array */
+static
+uint64_t array_max(uint64_t array[], unsigned int len) {
+	uint64_t max = 0;
+	int i;
+	for (i = 0; i < len; i++)
+		max = MAX(max, array[i]);
+
+	return (max);
+}
```

Comment:
It may make sense to move generic functions which don't take a `iostat_cbdata_t` as an argument in to `cmd/zpool/zpool_util.c`.